### PR TITLE
refactor(npm-scripts)

### PR DESCRIPTION
**Pull request by debloper**
Addresses #2376 
Also updates CI/CD pipeline to reflect the change.

- ~~Breaks docker-compose & minishift orchestration.~~ Fixed.
- ~~Breaks `run-planner.sh` script's watch mode.~~ Fixed

### CODE REVIEW

**Comment by debloper** [comment]
As part of this PR, I've updated semantic-release module and how it works now.
We were about 4 versions behind and completely on a different era of its CLI.

More details: https://github.com/semantic-release/semantic-release/pull/480

**Comment by debloper** [comment]
screenshot from 2018-01-16 14-15-48
This is how the updated docs on new npm scripts will look like.

**Comment by jarifibrahim** [comment]
@debloper `npm run build` doesn't work. The tests would obviously fail if planner doesn't start. 
This is what I see after building planner via `npm run build`
screenshot from 2018-01-17 13-43-07

**Comment by debloper** [comment]
@jarifibrahim which is why I'm consistently saying, the steps after functional tests on CI are more important for me. Considering putting the functional tests after the build step in CI - any objections?

**Comment by jarifibrahim** [comment]
@debloper No, I haven't done `npm link ../dist-watch`. 

> Considering putting the functional tests after the build step in CI - any objections?

The functional tests are already the last step in the CI build https://github.com/fabric8-ui/fabric8-planner/blob/master/deploy/release.groovy#L17
The issue here is that the files are not being copied in the correct directory.
`npm build` before/after the functional tests won't help in this case.

**Comment by debloper** [comment]
@jarifibrahim 

> No, I haven't done npm link ../dist-watch
Which directory did you link npm to from runtime then?

> The functional tests are already the last step in the CI build
No, they come before fabric8-ui build.

**Comment by michaelkleinhenz** [comment]
Does this still break run_planner.sh?

**Comment by debloper** [comment]
@jarifibrahim @rgarg1 

> `npm run build` doesn't work. The tests would obviously fail if planner doesn't start.

Sure it does: https://jenkins.cd.test.fabric8.io/blue/organizations/jenkins/fabric8-ui%2Ffabric8-planner/detail/PR-2399/13/pipeline/

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2399-30 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2399-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2399-32 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2399-fabric8-planner.badger.fabric8.io

**Comment by michaelkleinhenz** [comment]
Could be that this conflicts with https://github.com/fabric8-ui/fabric8-planner/pull/2419 as I also modified some of the stuff there.

**Comment by debloper** [comment]
> Could be that this conflicts with #2419 as I also modified some of the stuff there.

@michaelkleinhenz I don't think there's any functional conflict to happen for release.groovy. You might have to move a line or two up or down that is.

BTW, the build 33 failed with `npm ERR! Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?` is one off random error or we did something new that needs changes?

**Comment by fabric8cd** [comment]
@debloper fabric8/fabric8-ui:SNAPSHOT-PR-2399-36 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2399-fabric8-planner.badger.fabric8.io

File: package.json [hunk 1]
```diff
@@ -185,7 +180,7 @@
     "rimraf": "2.6.2",
     "run-sequence": "2.2.0",
     "script-ext-html-webpack-plugin": "1.8.7",
-    "semantic-release": "8.2.0",
+    "semantic-release": "12.2.2",
```
Comment (joshuawilson) [inline package.json]: I think I had a issue with the newer version of semantic-release. But that was a while ago, don't remember what was the cause.

Comment (debloper) [reply package.json]: Must have been the pre/post steps, I'd guess, which aren't required anymore.
I had to refactor the release step a little bit to make it work with our `build > release` routines.

**Pull request merged**